File: android/app/src/main/java/com/serene/logcal/model/PendingMealLog.kt
package com.serene.logcal.model

import android.net.Uri
import java.time.LocalDate

sealed class PendingLogStatus {
    object Processing : PendingLogStatus()
    data class Completed(val response: MealLogResponse, val entryId: String) : PendingLogStatus()
    data class Failed(val error: String) : PendingLogStatus()
}

data class PendingMealLog(
    val id: String,
    val foodText: String,
    val imageUris: List<Uri>,
    val mealType: MealType,
    val date: LocalDate,
    val createdAtMillis: Long = System.currentTimeMillis(),
    val isPreviewOnly: Boolean = false,
    val status: PendingLogStatus = PendingLogStatus.Processing,
) {
    val displayText: String
        get() {
            val trimmed = foodText.trim()
            if (trimmed.isNotEmpty()) return trimmed
            if (imageUris.isNotEmpty()) return "Photo meal"
            return "${mealType.rawValue.replaceFirstChar { it.uppercase() }} log"
        }
}

File: logcal/Models/PendingMealLog.swift
//
//  PendingMealLog.swift
//  logcal
//

import Foundation
import UIKit

enum PendingLogStatus: Equatable {
    case processing
    case completed(response: MealLogResponse, entryId: UUID)
    case failed(error: String)
}

struct PendingMealLog: Identifiable, Equatable {
    let id: UUID
    let foodText: String
    let images: [UIImage]
    let mealType: MealType
    let selectedDate: Date
    let createdAt: Date
    var isPreviewOnly: Bool = false
    var status: PendingLogStatus
    
    var displayText: String {
        let trimmed = foodText.trimmingCharacters(in: .whitespacesAndNewlines)
        if !trimmed.isEmpty {
            return trimmed
        }
        if !images.isEmpty {
            return "Photo meal"
        }
        return "\(mealType.rawValue.capitalized) log"
    }
    
    static func == (lhs: PendingMealLog, rhs: PendingMealLog) -> Bool {
        lhs.id == rhs.id && lhs.status == rhs.status && lhs.foodText == rhs.foodText && lhs.mealType == rhs.mealType
    }
}
